### PR TITLE
Using CPU pool to build Linux GPU C API Package

### PR DESCRIPTION
**Pull request by jchen351**

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/stages/jobs/linux-gpu-tensorrt-packaging-job.yml
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/stages/jobs/linux-gpu-tensorrt-packaging-job.yml:1]: Moving this back to tools/ci_build/github/azure-pipelines/stages/nuget-linux-cuda-packaging-stage.yml

File: tools/ci_build/github/azure-pipelines/stages/nuget-linux-cuda-packaging-stage.yml [hunk 1]
```diff
@@ -70,15 +70,81 @@ stages:
...
-  - template: jobs/linux-gpu-tensorrt-packaging-job.yml
-    parameters:
-      artifactName: 'onnxruntime-linux-x64-tensorrt-$(OnnxRuntimeVersion)'
-      artifactNameNoVersionString: 'onnxruntime-linux-x64-tensorrt'
-      buildJava: ${{ parameters.buildJava }}
-      buildJavaOption: '--build_java'
-      buildNodejs: ${{ parameters.buildNodejs }}
-      buildNodejsOption: '--build_nodejs'
-      CudaVersion: ${{ parameters.CudaVersion }}
+  - job: Linux_C_API_Packaging_TensorRT
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/stages/nuget-linux-cuda-packaging-stage.yml:73]: @yf711 fyi

File: tools/ci_build/github/azure-pipelines/stages/nuget-linux-cuda-packaging-stage.yml [hunk 2]
```diff
@@ -70,15 +70,81 @@ stages:
...
+    dependsOn: []
+    workspace:
+      clean: all
+    timeoutInMinutes: 180
+    pool: 'onnxruntime-Ubuntu2204-AMD-CPU'
```
Comment (yf711) [inline tools/ci_build/github/azure-pipelines/stages/nuget-linux-cuda-packaging-stage.yml:78]: GPU instance not required when building this task?

Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/stages/nuget-linux-cuda-packaging-stage.yml:79]: No, gpu is not needed when building, but it is required when testing it.

File: tools/ci_build/github/linux/build_cuda_c_api_package.sh [hunk 3]
```diff
@@ -1,9 +1,10 @@
 #!/bin/bash
 set -e -x
-docker run --gpus all -e NVIDIA_VISIBLE_DEVICES=all --rm --volume \
+docker run --rm --volume \
 $BUILD_SOURCESDIRECTORY:/onnxruntime_src --volume $BUILD_BINARIESDIRECTORY:/build \
 --volume /data/models:/build/models:ro --volume /data/onnx:/data/onnx:ro -e NIGHTLY_BUILD onnxruntimecuda${CUDA_VERSION_MAJOR}build \
 /usr/bin/python3.9 /onnxruntime_src/tools/ci_build/build.py --enable_lto --build_java --build_nodejs --build_dir /build --config Release \
 --skip_submodule_sync  --parallel --use_binskim_compliant_compile_flags --build_shared_lib --use_cuda --cuda_version=$CUDA_VERSION \
 --cuda_home=/usr/local/cuda-$CUDA_VERSION --cudnn_home=/usr/local/cuda-$CUDA_VERSION \
+--skip_tests \
```
Comment (jchen351) [inline tools/ci_build/github/linux/build_cuda_c_api_package.sh:9]: @jywu-msft  @yf711 is it OK to skill tests here? There is another test stage later called `Nuget_Test_Linux_GPU` stage and wonder if their test coverages are redundant?

By just skipping this test and using `onnxruntime-Ubuntu2204-AMD-CPU` to build CUDA packages will save whole pipeline build time by 1 hours+, down from 3 - 5 hours to 2 - 3 hours.

Comment (jchen351) [reply tools/ci_build/github/linux/build_cuda_c_api_package.sh:9]: It is also less expensive and wait time to get an AMD CPU machine than any Nvidia GPU machine.

**Review by snnn — APPROVED** [review]

**Review by yf711 — APPROVED** [review]

**Pull request merged**